src/api/hooks/useGetProposal.tsx

```diff
@@ -56,36 +57,50 @@ const fetchProposalMetadata = async (
 
   const response = await fetch(raw_metadata_location);
   // validate response status
-  if (response.status !== 200) return null;
+  if (response.status !== 200) {
+    return {
+      errorMessage: "Proposal metadata not found",
+    };
+  }
 
   const metadataText = await response.text();
-
-  //validate metadata
+  // validate metadata
   const metadata_hash = proposalData.metadata.data.find(
     (metadata) => metadata.key === "metadata_hash",
   )!.value;
-
   const hash = sha3_256(metadataText);
-  if (hex_to_string(metadata_hash) !== hash) return null;
+  if (hex_to_string(metadata_hash) !== hash) {
+    return {
+      errorMessage: "Metadata is invalid",
```

Comment:
what should this error text be?

Reply:
this is what was causing the error in the slack threads

Reply:
"Metadata hash mismatch" is probably more accurate here

Reply:
updated